packages/flutter/lib/src/foundation/collections.dart

```diff
@@ -355,3 +355,10 @@ void _merge<T>(
   target[targetOffset++] = secondElement;
   target.setRange(targetOffset, targetOffset + (secondEnd - cursor2), secondList, cursor2);
 }
+
+/// Casts the provided `map` [Object] to a typed `Map<S, T>` where `S` is the
+/// type of the keys and `T` is the type of the values.
+Map<S, T> castToMap<S, T>(Object map) {
+  final Map<dynamic, dynamic> casted = map as Map<dynamic, dynamic>;
```

Comment:
nit: document that the object has to be cast-able to a `Map<dynamic, dynamic>`, otherwise will throw?

Reply:
If this is any kind of map this actually will not throw until you attempt to access something. The example below does not throw unless you uncomment the print.

```
void main() {
  var intMap = <int, int>{2: 3};
  var stringMap = intMap.cast<String, String>();
 // print(stringMap.keys.first);
}
```
This also means the type checks and casts are incurred on every access, instead of once at the beginning. If you wanted something that threw eagerly, you could use Map.from instead

Reply:
Ahhh. Should we really be exposing this method then?

Reply:
Looking at this again, I don't need this method at all. I am removing it.